apps/web/src/pages/templates/workflow/workflow/node-types/WorkflowNode.tsx

```diff
@@ -119,7 +131,22 @@ export function WorkflowNode({
     }
 
     return true;
-  }, [integrations, tabKey, isEmailLimitReached, isSmsLimitReached]);
+  }, [integrations, tabKey, isEmailLimitReached, isSmsLimitReached, isChannelStep]);
+
+  const getPrimaryIntegration = useMemo(() => {
+    if (channelType === StepTypeEnum.IN_APP) {
+      return InAppProviderIdEnum.Novu;
+    }
```

Comment:
the `IN_APP` integration could be also "deactivated", so we should show the no provider icon

Reply:
also `SMS` and `EMAIL` integrations could be deleted

Reply:
Now it will return undefined, if there's no active integration found, hence handling the situation where the integration are deleted